text/66-self-managed-stack-sets.md

```diff
@@ -0,0 +1,537 @@
+Stack set instances are deployed independently from CDK context, and exist in different accounts/regions. 
+This is dicussed more in the design details.
+
+### What alternative solutions did you consider?
+
+An alternative is the usage of `cdk.CfnStackSet`, which is not ideal for customers with existing stack sets since there is significant migration effort.
+
+### What is the high level implementation plan?
+* Create StackSet core construct
+* Create StackSet artifact type in the Cloud Assembly
+* Create Bootstrapless StackSet Synthesizer
+* Update CLI to collect StackSet artifact types in addition to Stacks
+* Deployment code for creating/updating stack sets
+
+### Are there any open issues that need to be addressed later?
+
+* CDK asset handling
+* Future support for service managed stack sets
+* CDK Pipelines
+
+---
+# Implemenation Details
+
+# Requirements
+
+## In Scope Features
+
+* Create Self-Managed Stack Set
+  * This should create the stack set directly, as opposed to the [CfnStackSet](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-cloudformation.CfnStackSet.html) construct which is defined in a single CloudFormation stack that deploys the stack set
+* Update Self-Managed Stack Set
+  * Update includes updating **all existing** stack set instances 
+  * Set Operation Preference Configuration
+
+## Out of Scope Features
+
+* Delete Self-Managed Stack Set
+* Service Managed Stack Set
+* Add/Remove stack set instances through CDK
+* Update to only specific stack set instances
+* Stack Set Instance Level Parameter Overrides
+* CDK Assets (CDK File Assets/CDK Docker Assets)
+  * More information on CDK asset handling and complexities described in the [CDK Assets Handling](#4-cdk-assets-handling) section.
+
+## Assumptions
+
+* Parent account contains Stack Set administration role and child accounts contains Stack Set execution role for enabling stack set operations
+  * https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/stacksets-prereqs-self-managed.html
+  * The Stack Set administration role in the parent account is used by CloudFormation to assume the execution role in the child account. This role can be created manually or may be created through a CDK stack. This `role arn` is passed into the stack set CDK construct properties 
+  * The Stack Set execution role in the child is the role assumed by the administration role in the parent account, and utilized to provision resources in the child account. This role should be provisioned manually, and the `role name` is passed into the stack set CDK construct properties
+* Stack Set Instances
+  * In this initial release, stack set instances should be added manually to stack set after it was initially created by CDK
+  * Since this initial iteration does not include CDK assets support such as files/docker images, bootstrapping multiple regions is not required. Bootstrapping a single region in the parent account may be required for templates larger than 50kb. 
```

Comment:
Line length again.

Reply:
Not sure I understand what you mean by "Bootstrapping a single region"...? You will need to `cdk bootstrap` in every region you're creating a StackSet with a template > 50 KB, right?

Reply:
No, so when a stack set is created/update with a template (either inline or through S3), the CloudFormation service itself stores the template and manages the templates. I.e. stack set instances in child accounts don't read from the bootstap bucket

Reply:
I'm not talking about the StackSet instances though. I'm talking about StackSets themselves (the parent account).

Reply:
hmmm, if you deploy a stack set in different regions then yes. Generally, a user would only deploy a stack set in a single region which deploys stack set instances to different accounts/regions.

Reply:
Let's be precise then.

"The parent account needs to be bootstrapped with the `cdk bootstrap` command in each region that a StackSet is to be created in."

Or something like that. In general, the whole point of the RFC is to precisely describe the behavior and requirements of the new functionality.

Reply:
Talked offline on how we can use bootstrap for default stack set roles